src/assets/htmx/htmx.md

```diff
@@ -0,0 +1,374 @@
+
+First let's make clicking on the todo item go to a different page that will contain form for updating the todo item.
+For that, we need to get an anchor tag instead of a div with the todo text inside a div
+
+
+
+In the server i will get the task and return the proper form for updating that task.
+
+```go
+func main() {
+    ...
+    http.HandleFunc("/task/", indivisualTask)
+    ...
+}
+```
+
+**example response from server**
+
+```htmx
+  <div hx-boost="true">
+      <a href="/task/asdasda">asdasda</a>
+  </div>
+```
+
+Here the `hx-boost` attribute will replace the body element and add the previous page to the history entry which will allow you to go back if needed.
+
+![demo](src/assets/htmx/bktyFpm.gif)
+
+For the actual editing, it will be similar to the other examples.
+
+```html
+<div>
+  <div>clean room</div>
+  <form id="myForm2" hx-put="/edit" hx-target="#app">
+    <input type="text" name="task" required="" />
+    <input type="hidden" name="old" value="clean room" />
+    <input type="submit" value="edit" />
+  </form>
+</div>
+```
+
+Now let's perform deletion on the list.
+
+First, we have to make a delete button.
+
+For that, I will update the list template and add a delete form
+
+```html
+<div hx-boost="true" class="new-element">
+  <a href="/task/dasdas">dasdas</a>
+  <form
+    hx-trigger="click"
+    hx-delete="/delete"
+    hx-target="closest .new-element"
+    hx-swap="outerHTML swap:1s"
+  >
+    <input type="hidden" name="task" value="dasdas" />
+    delete
+  </form>
+</div>
+```
+
+Most of the snippets in the delete form are similar to the other examples. but in the `hx-swap="outerHTML swap:1s"` attribute I added an extra value . All that value does is delay the swapping of the element by 1 second because i want to do a css animation during that time.
+
+While htmx is doing the swapping of the element it also adds the `htmx-swapping` class to the element.
+
+Now i can add any css effect I want to the `htmx-swapping` class.
+
+```css
+.htmx-swapping {
+  opacity: 0;
+  transition: opacity 1s ease-out;
+}
+```
+
+![demo](src/assets/htmx/khdzEgM.gif)
+
+## WebSockets
+
+Currently, if we add a todo task to our list it won't be shown to another tab/browser opening the same page which can be achieved using websockets. Fortunately htmx supports websockets.
+Let's first make a connection to a websocket.
+
+```html
+<div id="ws" hx-ws="connect:/ws" hx-trigger="task"></div>
+```
+
+The `hx-ws` attribute takes the websocket endpoint and makes a connection to it.
+
+```go
+import(
+    ...
+    "golang.org/x/net/websocket"
+    "sync"
+    ...
+)
+
+type Server struct {
+    conns map[*websocket.Conn]bool
+    mutex sync.Mutex
+}
+
+func NewServer() *Server {
+    return &Server{
+        conns: make(map[*websocket.Conn]bool),
+        mutex: sync.Mutex{},
+    }
+}
+func main(){
+    ...
+    server := NewServer()
+    http.Handle("/ws", nil)
+    ...
+}
+```
```

Comment:
It might be useful for the reader to comment a little bit on what you're doing in this code snippet.